components/admin_console/billing/payment_info_details.tsx

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import {FormattedMessage} from 'react-intl';
+import {useSelector} from 'react-redux';
+
+import FormattedMarkdownMessage from 'components/formatted_markdown_message';
+import CardImage from 'components/payment_form/card_image';
+import {GlobalState} from 'types/store';
+
+export interface PaymentInfoDetailsProps {
+    children?: React.ReactNode;
+}
+
+const PaymentInfoDetails: React.FC<PaymentInfoDetailsProps> = ({children}: PaymentInfoDetailsProps) => {
+    const paymentInfo = useSelector((state: GlobalState) => state.entities.cloud.customer);
+
+    if (!paymentInfo?.payment_method && !paymentInfo?.billing_address) {
```

Comment:
It seems confusing seeing in the same statement the non-null assertion `!` but at the same time doing the optional chaining in case it's null. Maybe just: 

```suggestion
    if (paymentInfo?.payment_method && paymentInfo?.billing_address) {
```

Reply:
This is still pending, thoughts on this?

Reply:
the logic using the ! is for negating a condition, saying if NO payment method and NO billing address return null. But the thing is that before checking for the payment_method and the billing_address we need to make sure that the paymentInfo Object exists and is not null, otherwise JS will thrown an error because trying to access a property on a non existing object. The alternative to not using the ! in conjunction with the ? would be to do this same logic as:

`if (paymentInfo && (!paymentInfo.payment_method && !paymentInfo.billing_address)`

Reply:
That is the purpose of the [optional chaining](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html) operator `?.` it will stop running if the object that is applied to is null or undefined so it's the same as the expression above.